db/db.js

```diff
@@ -0,0 +1,53 @@
+const db = require('../models/index.js').sequelize;
+
+const { Sequelize } = db;
+
+const Movie = require('../models/movie.js')(db, Sequelize);
+const Showtime = require('../models/showtime.js')(db, Sequelize);
+const Theater = require('../models/theater.js')(db, Sequelize);
+
+Movie.hasMany(Showtime, { foreignKey: 'movie_id', sourceKey: 'id' });
+Theater.hasMany(Showtime, { foreignKey: 'theater_id', sourceKey: 'id' });
+
+// get the closest theater
+// current definition of 'closest': Math.abs(theaterZip - inputZip)
+// reference of sequelize.fn('abs') https://github.com/sequelize/sequelize/issues/2657
+// can improve this query by adding a GMap API call in the future
+module.exports.getTheater = (zip, callback) => {
+  Theater.findAll({
```

Comment:
It looks like the sequelize.fn and sequelize.condition doesn't work on the .findOne.
It's a bit of hack here that I am determining the closest theater by calculating the closest zipcode number. It will be the next step (stretch goal) for me to actually do a GMap integration instead of the current maths.